### PR TITLE
Dance AI: Add start over button

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Adds a "Start Over" icon button to the top left corner. As per the mocks, this is present on the results page, and disabled on the explanation page. Does not appear on any other pages in the modal.

Note that we may end up changing the icon (I personally found it a bit confusing at first, but also understand that it needs to contract with start over) but that change should just be a matter of changing the icon class name(s).

Also made a few related quality of life changes as I noticed them
- Extracted conditional button logic into a separate `ModalButton` component - this might be a bit verbose, but it gives us a way to more cleanly conditionally show/hide/disable buttons based on the current modal mode. Happy to discuss though if folks feel that it would be better to keep the logic inside DanceAiModal.
- Moved icons and text that were previously component children into Button props for consistency
- Added `aria-label`s for icon buttons

Results pane:
Screenshot 2023-11-14 at 11 54 29 AM

Voiceover readout for Start Over button:
Screenshot 2023-11-14 at 11 55 58 AM

Voiceover readout for Explanation button:
Screenshot 2023-11-14 at 11 56 18 AM

RTL Results pane:
Screenshot 2023-11-14 at 3 49 42 PM

## Links

https://trello.com/c/PEcuk8tz

## Testing story

Tested locally.

### CODE REVIEW

File: apps/src/dance/ai/DanceAiModal.tsx [hunk 1]
```diff
@@ -818,53 +819,68 @@ const DanceAiModal: React.FunctionComponent = () => {
...
+          </div>
+          <div id="buttons-area-top-right">
+            <ModalButton
+              currentMode={mode}
+              showFor={[Mode.RESULTS]}
+              id="explanation-button"
+              onClick={handleExplanationClick}
+              color={Button.ButtonColor.neutralDark}
+              className={moduleStyles.button}
+              aria-label={i18n.danceAiModalExplanationButton()}
```
Comment (sanchitmalhotra126) [inline apps/src/dance/ai/DanceAiModal.tsx:843]: @breville just confirming this was the intended label for this button

Comment (breville) [reply apps/src/dance/ai/DanceAiModal.tsx:843]: Yes, I believe so.

**Comment by breville** [comment]
Let's just hide the "Start over" button on the explanation pane, since we also hide all the other buttons.

**Comment by breville** [comment]
Should we have the icon-only buttons show their alt-text on hover?

**Comment by sanchitmalhotra126** [comment]
> Should we have the icon-only buttons show their alt-text on hover?

Oh good call, will update.

**Review by breville — APPROVED** [review]
Ran locally and looks good.  Love the new button component!

**Review by levadadenys — APPROVED** [review]
LGTM!
Nice idea to add an optional title so that icon only button can have a tooltip, thanks!

**Pull request merged**